test/tests/mortar/continuity-3d-non-conforming/continuity_penalty_sphere_hex20.i

```diff
@@ -0,0 +1,126 @@
+[Mesh]
+  second_order = true
+  [file]
+    type = FileMeshGenerator
+    file = spheres_hex20.e
+  []
+  [secondary]
+    input = file
+    type = LowerDBlockFromSidesetGenerator
+    new_block_id = 11
+    new_block_name = "secondary"
+    sidesets = '101'
+  []
+  [primary]
+    input = secondary
+    type = LowerDBlockFromSidesetGenerator
+    new_block_id = 12
+    new_block_name = "primary"
+    sidesets = '102'
+  []
+[]
+
+[Problem]
+  kernel_coverage_check = false
+[]
+
+[Variables]
+  [T]
+    block = '1 2'
+    family = LAGRANGE
+    order = SECOND
+  []
+[]
+
+[BCs]
+  [neumann]
+    type = FunctionGradientNeumannBC
+    exact_solution = exact_soln_primal
+    variable = T
+    boundary = '1 2'
+  []
+[]
+
+[Kernels]
+  [conduction]
+    type = Diffusion
+    variable = T
+    block = '1 2'
+  []
+  [sink]
+    type = Reaction
+    variable = T
+    block = '1 2'
+  []
+  [forcing_function]
+    type = BodyForce
+    variable = T
+    function = forcing_function
+    block = '1 2'
+  []
+[]
+
+[Functions]
+  [forcing_function]
+    type = ParsedFunction
+    value = 'x^2 + y^2 + z^2 - 6'
+  []
+  [exact_soln_primal]
+    type = ParsedFunction
+    value = 'x^2 + y^2 + z^2'
+  []
+  [exact_soln_lambda]
+    type = ParsedFunction
+    value = '4'
+  []
+[]
+
+[Debug]
+  show_var_residual_norms = 1
+[]
+
+[Constraints]
+  [mortar]
+    type = PenaltyEqualValueConstraint
+    primary_boundary = 2
+    secondary_boundary = 1
+    primary_subdomain = '12'
+    secondary_subdomain = '11'
+    secondary_variable = T
+    penalty_value = 1.0e5
+    correct_edge_dropping = true
+  []
+[]
+
+[Preconditioning]
+  [smp]
+    type = SMP
+    full = true
+  []
+[]
+
+[Executioner]
+  solve_type = NEWTON
+  type = Steady
+  petsc_options_iname = '-pc_type -snes_linesearch_type -pc_factor_shift_type '
+                        '-pc_factor_shift_amount'
+  petsc_options_value = 'lu       basic                 NONZERO               1e-15'
```

Comment:
Can you do `boomeramg` with this problem? I'm curious since we no longer have a saddle point

Reply:
Yeah, we can do BoomerAMG with the penalty tests. Using the same BoomerAMG options for the LM-based enforcement doesn't make these small tests converge. So, I guess things are making sense.